test/unit/core/type_traits/basic_test.cpp

```diff
@@ -34,3 +38,73 @@ TEST(type_trait, remove_cvref_t)
     EXPECT_FALSE((std::is_same_v<int*, seqan3::remove_cvref_t<int[3]>>));   // type stays same
     // the last example would be true for std::decay_t
 }
+
+//------------------------------------------------------------------------------
+// Tests for is_constexpr
+//------------------------------------------------------------------------------
+
+constexpr
+int    constexpr_nonvoid_free_fun(int i) { return i; }
+int nonconstexpr_nonvoid_free_fun(int i) { return i; }
+
+constexpr
+int constexpr_nonvoid_free_fun_const_ref(int const & i) { return i; }
+int nonconstexpr_nonvoid_free_fun_const_ref(int const & i) { return i; }
+
+constexpr
+void    constexpr_void_free_fun(int) { return; }
+void nonconstexpr_void_free_fun(int) { return; }
+
+struct constexpr_nonvoid_member_t
+{
+    int constexpr get_i(int i)
+    {
```

Comment:
since you were putting a small function scope on the same line as the function declaration above, I would also do it here to be consistent (or change the above)

Reply:
Ok I looked a the code. And I decided not to change either of it. The class defintion look more like the riht style guide and that is good. The free function definitons above would really bloat the test and do not help for understanding what is happening. Since it is a test and we don't enforce strict rules on tests this should stay as is.